src/com/nkita/practice/collections/PersonStatisticHandler.java

```diff
@@ -176,27 +176,67 @@ public void checkFullStatisticWithoutDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         System.out.println("Statistics with duplicate");
 
-        for (Person person : persons) {
-            personSet.add(person);
-            System.out.println(personSet);
-            return;
+        for (Person person : personSet) {
+            System.out.println(person);
         }
 
 
     }
 
 
-
     public void removeDuplicates() {
         Set<Person> personSet = new LinkedHashSet<>(persons);
         persons.clear();
         persons.addAll(personSet);
 
-        }
-
-
     }
 
 
+    public void checkOldestPerson(Gender femaleAndMale) {
+        List<Person> genderMaleAgeAndGenderFemaleAge = new ArrayList<>();
+
+        int max = 0;
+        if (femaleAndMale == Gender.FEMALE) {
```

Comment:
Now it is better.
BUT now you have many statements nested inside each other
You check gender with an if - else statement, then create "foreach" loops, and then in each loop you have many statements to check the gender again. This is not good.
Instead of that you can only create one "foreach" loop
and perform all necessary check in it to find max age. And then you can use logic from method "checkStatisticByAge".